packages/core/src/Radio/Radio.tsx

```diff
@@ -78,7 +80,18 @@ const Radio: React.FC<IRadioProps> = React.forwardRef((props, ref) => {
   const borderAdjustedSize = size + 4
 
   return (
-    <RadioWrap color={props.color} checked={checked} disabled={disabled}>
+    <RadioWrap
+      color={props.color}
+      checked={checked}
+      disabled={disabled}
+      data-name={
+        (checked && disabled) || (!checked && disabled)
+          ? 'disabled'
+          : checked && !disabled
+          ? 'checked'
+          : 'unchecked'
```

Comment:
n-b This may be easier to read as a function with `if/else` statements to avoid the nested ternary.